mdpow/workflows/dihedrals.py

```diff
@@ -549,25 +633,28 @@ def plot_violins(df, resname, figdir=None, molname=None, width=0.9, solvents=SOL
         newdir = os.path.join(figdir, subdir)
         os.mkdir(newdir)
 
-    section = df.groupby(by='selection')
+    section = df.groupby(by="selection")
+
+    plot_pdf_width_px = plot_pdf_width * 3.7795275591
 
     for name in section:
-        plot = dihedral_violins(name[1], width=width, solvents=solvents)
-        plot.set_titles(f'{molname},{name[0]}, ''{col_name}')
-        # plot.set_titles needs to stay here during future development
-        # this locale ensures that plots are properly named,
-        # especially when generated for a projecct iteratively
+        
+        fig = build_svg(mol=mol, molname=molname, name=name, ab_pairs=ab_pairs,
+                        solvents=solvents, width=width)
+
+        figfile = pathlib.Path(newdir) / f"{molname}_{name[0]}_violins.pdf"
+        plot_pdf = cairosvg.svg2pdf(bytestring=fig.tostr(), write_to=str(figfile),
+                                    output_width=plot_pdf_width_px)
+
```

Comment:
Might be worthwhile closing the fig here at the end to conserve resources
```
fig.close()
```
It also makes clearer that the figures are transient.

Reply:
can you close an `svgutils` figure object like that?

Reply:
I don't know — I *assumed* `fig` was a matplotlib Figure instance.

Reply:
@orbeckst

Can we use something like `fig.close()` somewhere to suppress the default `matplotlib` output (every MPL figure object created) when working in notebooks in a way that will not impact building the SVG figures from the MPL figure objects (which are not saved, but converted to SVG from memory)? I have not checked this in something like iPython, but in Jupyter you will get as output all of the figures before they are converted to SVG. This means the output just has a blank space where the RDKit Mol object is supposed to be. 

After the MPL figure is converted to SVG it is combined with the SVG render of the highlighted/labeled RDKit Mol object, then converted to and saved as a PDF file. Then, all of the PDF files are combined and additionally saved as one PDF file that contains all of the final figures for that dihedral atom group. I think the problem is that as matplotlib generates the figures it send them to output, but I do not know if closing them will remove them from memory and interfere with downstream processes. 

The incorrect output is not a pertinent issue, but it would be cool if there was a way to eliminate this without affecting subsequent processes. Instead of closing the figures, I believe there is a way to suppress the output, I just have not been able to figure it out yet.

Reply:
Do you mean something like https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.ioff.html ?

I am pretty sure that `fig.close()` removes the in-memory object. In fact, rendering a figure is a separate process from creating the figure object itself.